R/StackedLearner_helpers.R

```diff
@@ -0,0 +1,336 @@
+### other helpers ###
+
+#' Sets the predict.type for the super learner of a stacked learner
+#' @param learner StackedLearner.
+#' @param predict.type "prob" or "response".
+#' @export
+
+setPredictType.StackedLearner = function(learner, predict.type) {
+  lrn = setPredictType.Learner(learner, predict.type)
+  lrn$predict.type = predict.type
+  if ("super.learner" %in% names(lrn)) lrn$super.learner$predict.type = predict.type
+  return(lrn)
+}
+
+#' Returns response from Prediction object 
+#' @param pred Prediction
+#' @param full.matrix Wether all n prediction values should be returned or in case of binary classification only one 
+
+getResponse = function(pred, full.matrix = NULL) {
+  # if classification with probabilities
+  if (pred$predict.type == "prob") {
+    if (full.matrix) {
+      # return matrix of probabilities
+      td = pred$task.desc
+      predReturn = pred$data[, paste("prob", td$class.levels, sep = ".")]
+      colnames(predReturn) = td$class.levels
+      return(predReturn)
+    } else {
+      # return only vector of probabilities for binary classification
+      return(getPredictionProbabilities(pred))
+    }
+  } else {
+    # for regression case
+    pred$data$response
+  }
+}
+
+#' Returns response or probabilites from Prediction with the speciality that the 
+#' first feature of a multiclass classification prediction will be removed to 
+#' overcome multicollinearity problems  
+#' @param pred Prediction from predict or resample
+
+getPredictionDataNonMulticoll = function(pred) {
+  if (any(class(pred) == "ResampleResult")) {
+    pred = pred$pred
+  }
+  pt = pred$predict.type
+  td = pred$task.desc
+  # for prediction from predict
+  #if (!is.null(pred$predict.type)) {
+  #  pt = pred$predict.type
+  #  td = pred$task.desc
+  #} else { # from predictions from resample
+  #  pt = pred$pred$predict.type
+  #  td = pred$pred$task.desc
+  #}
+  # if classification with probabilities
+  if (pt == "prob") {
+      pred.matrix = pred$data[, paste("prob", td$class.levels, sep = ".")]
+      colnames(pred.matrix) = td$class.levels
+      pred.matrix = pred.matrix[, -1, drop = TRUE] # 
+      return(pred.matrix)
+  } else {
+    # for perdict.type = "response"
+    pred$data$response
```

Comment:
and also maybe here `getPredictionResponse` check if there are other places where this can be replaced. In general
`$par.set` -> `getParamSet`
`$par.vals` -> `getHyperPars`
`$task.desc$id` -> `getTaskId`
Maybe also `getPredictionProbabilities`, ...